openstack/ussuri/ovs-kernel/os-vif.patch

```diff
@@ -0,0 +1,125 @@
+diff --git a/vif_plug_ovs/constants.py b/vif_plug_ovs/constants.py
+index 1eece80..3d9794b 100644
+--- a/vif_plug_ovs/constants.py
++++ b/vif_plug_ovs/constants.py
+@@ -24,5 +24,9 @@ PLATFORM_WIN32 = 'win32'
+ 
+ OVS_DPDK_INTERFACE_TYPE = 'dpdk'
+ 
++OVS_VDPA_TYPE = 'dpdkvdpa'
++OVS_VDPA_PREFIX = 'vdp'
++
+ # Neutron dead VLAN.
+ DEAD_VLAN = 4095
++
+diff --git a/vif_plug_ovs/linux_net.py b/vif_plug_ovs/linux_net.py
+index 20baab3..c80053d 100644
+--- a/vif_plug_ovs/linux_net.py
++++ b/vif_plug_ovs/linux_net.py
+@@ -32,6 +32,7 @@ from oslo_utils import excutils
+ from vif_plug_ovs import constants
+ from vif_plug_ovs import exception
+ from vif_plug_ovs import privsep
++from ovs_module.containerovsdb import ovsdb_lib as containerovsdb_lib
+ 
+ LOG = logging.getLogger(__name__)
+ 
+@@ -62,6 +63,21 @@ def _update_device_mtu(dev, mtu):
+ 
+ 
+ @privsep.vif_plug.entrypoint
++def create_vdpa_port(vif_name, pf_pci, vf_pci, vf_num, vdpa_socket_path):
++    container_ovs = containerovsdb_lib.BaseOVS(connection='tcp:127.0.0.1:6000')
++    container_ovs.create_ovs_vif_port("br0-ovs", vif_name,
```

Comment:
Feels like these parameters should really go into os-vif config opts as they depend on how the container was built/started.